src/goto-harness/recursive_initialization.cpp

```diff
@@ -797,3 +801,54 @@ bool recursive_initializationt::needs_freeing(const exprt &expr) const
   }
   return true;
 }
+
+code_blockt recursive_initializationt::build_function_pointer_constructor(
+  const exprt &result)
+{
+  PRECONDITION(can_cast_type<pointer_typet>(result.type()));
+  const auto &result_type = to_pointer_type(result.type());
+  PRECONDITION(can_cast_type<pointer_typet>(result_type.subtype()));
+  const auto &function_pointer_type = to_pointer_type(result_type.subtype());
+  PRECONDITION(can_cast_type<code_typet>(function_pointer_type.subtype()));
+  const auto &function_type = to_code_type(function_pointer_type.subtype());
+
+  std::vector<symbol_exprt> targets;
+
+  for(const auto &sym : goto_model.get_symbol_table())
+  {
+    if(sym.second.type == function_type)
+    {
+      targets.push_back(sym.second.symbol_expr());
+    }
+  }
+
+  code_blockt body{};
+
+  const auto function_pointer_selector =
+    get_fresh_local_symexpr("function_pointer_selector");
+  body.add(
+    code_assignt{function_pointer_selector,
+                 side_effect_expr_nondett{function_pointer_selector.type()}});
+  auto function_pointer_index = 0;
```

Comment:
`std::size_t` .. this way the deduced type is `int`.

Reply:
Done.